Adding a preference menu for changing font size

Wirdal opened the pull request:
I added options to change the font size for the answer and question boxes. I also cleaned up some of the commenting and spacing in the widgets function.

The sizes I added are 8-20, incremented by 2.

I was going to add options as well to change some of the coloring, as well as save the students email between quizzes, but I would see how this went over.



I have tested it by doing a dry run and it looks OK on my end, but I would appreciate a test yourself.

Commit:
Prelim work

Commit:
Added options for changing the font size

matloff commented:
This is really good.  Actually, last year I did reduce the font size,
but the TA complained it was too small.  Good to give the user a choice.

As I may have explained to you before, I am very, very slow to make
changes to OMSI, for fear that one little change that seems isolated
suddenly messes something else up when the exam comes.  In this case,
since you just changed the client, it should be OK.

My TAs for 132 and 145 are currently making another change.  After they
do, let's implement yours.

Thanks.

Norm

On Thu, Feb 14, 2019 at 04:05:54PM -0800, Chase Maguire wrote:
> I added options to change the font size for the answer and question boxes. I also cleaned up some of the commenting and spacing in the widgets function.
> 
> The sizes I added are 8-20, incremented by 2.
> 
> I was going to add options as well to change some of the coloring, as well as save the students email between quizzes, but I would see how this went over.
> 
> 
> 
> I have tested it by doing a dry run and it looks OK on my end, but I would appreciate a test yourself.
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/matloff/omsi/pull/7
> 
> -- Commit Summary --
> 
>   * Prelim work
>   * Added options for changing the font size
> 
> -- File Changes --
> 
>     M .gitignore (2)
>     M OmsiGui.py (43)
> 
> -- Patch Links --
> 
> https://github.com/matloff/omsi/pull/7.patch
> https://github.com/matloff/omsi/pull/7.diff
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/matloff/omsi/pull/7

Wirdal commented:
That works, I will work a bit more on some preferences, and preference saving.

Commit:
Added config support

Wirdal commented:
I've added support for a configuration file to save the users email and font selection through runs of the program.

I added a library, ConfigParser, to help with the .ini file that is also added.

I think a function could be added that updates the config files, but I didn't want to make that many changes, so it is done in-line for now.

Let me know what you think.

Commit:
Removing my info from the config
